core/domain/question_services.py

```diff
@@ -217,3 +228,319 @@ def update_question(
+    """Saves a QuestionRights domain object to the datastore.
+
+    Args:
+        question_rights: QuestionRights. The rights object for the given
+            question.
+        committer_id: str. ID of the committer.
+        commit_message: str. Descriptive message for the commit.
+        commit_cmds: list(QuestionChangeDict). A list of commands
+            describing what kind of commit was done.
+    """
+
+    model = question_models.QuestionRightsModel.get(
+        question_rights.id, strict=False)
+
+    model.manager_ids = question_rights.manager_ids
+    commit_cmd_dicts = [commit_cmd.to_dict() for commit_cmd in commit_cmds]
+    model.commit(committer_id, commit_message, commit_cmd_dicts)
+
+
+def create_new_question_rights(question_id, committer_id):
+    """Creates a new question rights object and saves it to the datastore.
+
+    Args:
+        question_id: str. ID of the question.
+        committer_id: str. ID of the committer.
+    """
+    question_rights = question_domain.QuestionRights(question_id, [])
+    commit_cmds = [{'cmd': CMD_CREATE_NEW}]
+
+    question_models.QuestionRightsModel(
+        id=question_rights.id,
+        manager_ids=question_rights.manager_ids
+    ).commit(committer_id, 'Created new question rights', commit_cmds)
+
+
+def get_question_rights(question_id, strict=True):
+    """Retrieves the rights object for the given question.
+
+    Args:
+        question_id: str. ID of the question.
+        strict: bool. Whether to fail noisily if no question rights with a
+            given id exists in the datastore.
+
+    Returns:
+        QuestionRights. The rights object associated with the given question.
+
+    Raises:
+        EntityNotFoundError. The question with ID question_id was not
+            found in the datastore.
+    """
+
+    model = question_models.QuestionRightsModel.get(question_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_question_rights_from_model(model)
+
+
+def check_can_edit_question(user_id, question_id):
+    """Checks if the user can edit the given question or not
+
+    Args:
+        user_id: str. The user ID of the user.
+        question_id: str. The question ID of the question.
+
+    Returns:
+        bool. Represents if the user can edit the question or not.
+    """
+    question_summary = get_question_summary_by_question_id(
+        question_id, strict=False)
+
+    if question_summary.status == feconf.QUESTION_STATUS_PENDING:
+        return False
+
+    if (user_services.is_admin(user_id) or
+            user_services.is_topic_manager(user_id)):
+        if question_summary.status == feconf.QUESTION_STATUS_APPROVED:
+            return True
+        return False
+
+    if (question_summary.status == feconf.ACTIVITY_STATUS_PRIVATE or
+            question_summary.status == feconf.QUESTION_STATUS_REJECTED):
+        return True
+
+    return False
+
+
+def publish_question(question_id, committer_id):
```

Comment:
Shouldn't this be approve_question or maybe approve_and_publish_question, since all approved questions are automatically published?

Reply:
approve and publish sounds good